chain/block.go

```diff
@@ -35,12 +39,36 @@ var (
 	_ block.StateSummary      = &SyncableBlock{}
 )
 
+type TrieItem struct {
+	x []byte
+}
+
+// CalculateHash hashes the values of a TestContent
+func (t TrieItem) CalculateHash() ([]byte, error) {
+	h := sha256.New()
+	if _, err := h.Write([]byte(t.x)); err != nil {
+		return nil, err
+	}
+
+	return h.Sum(nil), nil
+}
+
+// Equals tests for equality of two Contents
+func (t TrieItem) Equals(other merkletree.Content) (bool, error) {
+	otherTI, ok := other.(TrieItem)
+	if !ok {
+		return false, errors.New("value is not of type TestContent")
+	}
+	return bytes.Equal(t.x, otherTI.x), nil
+}
+
 type StatefulBlock struct {
 	Prnt   ids.ID `json:"parent"`
 	Tmstmp int64  `json:"timestamp"`
 	Hght   uint64 `json:"height"`
 
-	Txs []*Transaction `json:"txs"`
+	Txs       []*Transaction `json:"txs"`
+	BlockHash []byte         `json:"txsRoot"`
```

Comment:
nit: this should be `TxsRoot` not `BlockHash`. The hash of any block is returned by `ID()`.